cpg_workflows/stages/large_cohort.py

```diff
@@ -279,9 +279,13 @@ class DenseBackground(CohortStage):
     """
 
     def expected_outputs(self, cohort: Cohort) -> dict[str, Path]:
+        access_level = config_retrieve(['workflow', 'access_level'])
         background_datasets: list[str] = config_retrieve(['large_cohort', 'pca_background', 'datasets'], False)
         background_storage_paths: dict[str, Path] = {
-            dataset: Path(config_retrieve(['storage', dataset, 'default'])) for dataset in background_datasets
+            dataset: to_path(
+                config_retrieve(['storage', ('-').join(dataset.split(f'-{access_level}')[:-1]), 'default']),
+            )
+            for dataset in background_datasets
```

Comment:
I'm not sure I understand this... Well, I do, but I think there's a better solution. 

In the `large_cohort.pca_background.datasets` part of the config you've provided `['thousand-genomes-test']`, and later on there's `large_cohort.pca_background.thousand-genomes-test`. 

You can just coordinate this in config - if you list `['thousand-genomes']` as the dataset, and `large_cohort.pca_background.thousand-genomes` as the config section there's no need for this splitting. You'd have a separate config file for main and test modes, each pointing to the relevant input tables.